app/main.py

```diff
@@ -1,2 +1,70 @@
-class Dictionary:
-    pass
+from typing import Any, Hashable
+
+
+class Dictionary(object):
+    def __init__(self, capacity: int = 8, load_factor: float = 0.66) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.length = 0
+        self.limit = capacity * load_factor
+        self.hash_table = [[] for _ in range(capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length > self.limit:
+            self.resize()
+
+        index = self._get_index(key)
+
+        if self.hash_table[index]:
+            self.hash_table[index] = (key, hash(key), value, )
+        else:
+            self.hash_table[index] = (key, hash(key), value, )
+            self.length += 1
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._get_index(key, False)
+
+        if not self.hash_table[index]:
+            raise KeyError(key)
+        elif self.hash_table[index][0] == key:
+            return self.hash_table[index][2]
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._get_index(key, False)
+        bucket = self.hash_table[index]
+
+        if len(bucket) == 0 or bucket[0] != key:
+            raise KeyError(key)
+        else:
+            self.hash_table[index] = []
+            self.length -= 1
```

Comment:
Will this method work correctly if there was a collision with keys and you have deleted one of them?)